utils/ghc-pkg/Main.hs

```diff
@@ -622,7 +643,12 @@ getPkgDatabases verbosity mode use_user use_cache expand_vars my_flags = do
         [] -> do mb_dir <- getBaseDir
                  case mb_dir of
                    Nothing  -> die err_msg
-                   Just dir -> do
+                   Just dir' -> do
+                     mt <- getTarget my_flags
+                     dir <- case mt of
+                              Nothing -> pure dir'
+                              Just target -> pure (dir' </> "targets" </> target </> "lib")
```

Comment:
This path should be kept in sync with ghc: perhaps add a helper function where `getBaseDir` is? If it's not possible because of staging, some comment would be useful to remember this.

Reply:
The code in GHC is a bit different:

https://github.com/stable-haskell/ghc/blob/1c9db51407f2d8d5597e4d7a4cb0fdfd6778e04a/ghc/Main.hs#L142-L161

What did you have in mind?

Reply:
I was thinking about something like `getTargetDir :: FilePath -> String -> IO FilePath` taking the topdir and the target name and returning the target dir.